cli/tools/test/mod.rs

```diff
@@ -268,43 +268,77 @@ pub enum TestFailure {
   HasSanitizersAndOverlaps(IndexSet<String>), // Long names of overlapped tests
 }
 
-impl ToString for TestFailure {
-  fn to_string(&self) -> String {
+impl std::fmt::Display for TestFailure {
```

Comment:
For reviewers: this is just a refactor; `ToString` trait is automatically implemented for all types that implement `std::fmt::Display`, so basically it's recommended to implement `std::fmt::Display`